src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -71,38 +76,97 @@ public Task FindReferencesAsync(ISymbol symbol, CancellationToken cancellationTo
     public async Task FindReferencesAsync(
         ImmutableArray<ISymbol> symbols, CancellationToken cancellationToken)
     {
-        var unifiedSymbols = new MetadataUnifyingSymbolHashSet();
-        unifiedSymbols.AddRange(symbols);
+        var channel = Channel.CreateUnbounded<Reference>(s_channelOptions);
 
-        await _progress.OnStartedAsync(cancellationToken).ConfigureAwait(false);
-        try
+        await Task.WhenAll(
+            FindAllReferencesAndWriteToChannelAsync(),
+            ReadReferencesFromChannelAndReportToCallbackAsync()).ConfigureAwait(false);
```

Comment:
@ToddGrun how would you feel about extracting a helper somewhere to encode thsi pattern.  Specifically, the channel, and the two high level heleprs for writing to it and reading from it?

Reply:
Definitely agree that this would be useful. I can throw something together, but probably not until Monday.